server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3399,6 +3407,50 @@ else if (!template.isPublicTemplate() && caller.getType() != Account.ACCOUNT_TYP
                 hypervisorType, true, cmd.listInReadyState(), permittedAccounts, caller, listProjectResourcesCriteria, tags, showRemovedISO, null, null);
     }
 
+    @Override
+    public DetailOptionsResponse listDetailOptions(final ListDetailOptionsCmd cmd) {
+        final ResourceObjectType type = cmd.getResourceType();
+        final String resourceUuid = cmd.getResourceId();
+        final Map<String, List<String>> options = new HashMap<>();
+        switch (type) {
+            case Template:
+            case UserVm:
+                HypervisorType hypervisorType = HypervisorType.None;
+                if (!Strings.isNullOrEmpty(resourceUuid) && ResourceObjectType.Template.equals(type)) {
```

Comment:
shouldn't resourceUuid be validated in the cmd already?

Reply:
Defensive programming, we shouldn't assume; even if upper layers may have validated it.